plugins/node/opentelemetry-instrumentation-undici/src/undici.ts

```diff
@@ -0,0 +1,460 @@
+  // create the span and populate some atttributes, then link the span to the request for further
+  // span processing
+  private onRequestCreated({ request }: RequestMessage): void {
+    // Ignore if:
+    // - instrumentation is disabled
+    // - ignored by config
+    // - method is 'CONNECT'
+    const config = this._getConfig();
+    const shouldIgnoreReq = safeExecuteInTheMiddle(
+      () =>
+        !config.enabled ||
+        request.method === 'CONNECT' ||
+        config.ignoreRequestHook?.(request),
+      e => e && this._diag.error('caught ignoreRequestHook error: ', e),
+      true
+    );
+
+    if (shouldIgnoreReq) {
+      return;
+    }
+
+    const startTime = hrTime();
+    const rawHeaders = request.headers.split('\r\n');
+    const reqHeaders = new Map(
+      rawHeaders.map(h => {
+        const sepIndex = h.indexOf(':');
+        const name = h.substring(0, sepIndex).toLowerCase();
+        const val = h.substring(sepIndex + 1).trim();
+        return [name, val];
+      })
+    );
+
+    const requestUrl = new URL(request.origin + request.path);
+    const urlScheme = requestUrl.protocol.replace(':', '');
+    const attributes: Attributes = {
+      [SemanticAttributes.HTTP_REQUEST_METHOD]: request.method,
+      [SemanticAttributes.URL_FULL]: requestUrl.toString(),
+      [SemanticAttributes.URL_PATH]: requestUrl.pathname,
+      [SemanticAttributes.URL_QUERY]: requestUrl.search,
+      [SemanticAttributes.URL_SCHEME]: urlScheme,
+    };
+
+    const schemePorts: Record<string, string> = { https: '443', http: '80' };
+    const serverAddress = requestUrl.hostname;
+    const serverPort = requestUrl.port || schemePorts[urlScheme];
+
+    attributes[SemanticAttributes.SERVER_ADDRESS] = serverAddress;
+    if (serverPort && !isNaN(Number(serverPort))) {
+      attributes[SemanticAttributes.SERVER_PORT] = Number(serverPort);
+    }
+
+    const userAgent = reqHeaders.get('user-agent');
+    if (userAgent) {
+      attributes[SemanticAttributes.USER_AGENT_ORIGINAL] = userAgent;
+    }
+
+    // Get attributes from the hook if present
+    const hookAttributes = safeExecuteInTheMiddle(
+      () => config.startSpanHook?.(request),
+      e => e && this._diag.error('caught startSpanHook error: ', e),
+      true
+    );
+    if (hookAttributes) {
+      Object.entries(hookAttributes).forEach(([key, val]) => {
+        attributes[key] = val;
+      });
+    }
+
+    // Check if parent span is required via config and:
+    // - if a parent is required but not present, we use a `NoopSpan` to still
+    //   propagate context without recording it.
+    // - create a span otherwise
+    const activeCtx = context.active();
+    const currentSpan = trace.getSpan(activeCtx);
+    let span: Span;
+
+    if (config.requireParentforSpans && !currentSpan) {
+      span = trace.wrapSpanContext(INVALID_SPAN_CONTEXT);
+    } else {
+      span = this.tracer.startSpan(
+        `HTTP ${request.method}`,
```

Comment:
Per https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name this is meant to just be `{method}` ... and if that request.method isn't one of the upper-case well-known methods where the `http.request.method` was defaulted to `_OTHER`, then the string 'HTTP' is meant to be used.

Reply:
thanks for pointing this out. I've also added a test :)